cmake/HelperFunctions.cmake

```diff
@@ -6,7 +6,7 @@ function(generate_yaml)
 
     file(TIMESTAMP "${PROJECT_SOURCE_DIR}/plugins/plugins.yaml" PROFILE_FILE_TIMESTAMP "%s" UTC)
 
-    if(DEFINED CACHE{LAST_YAML_BUILD} AND DEFINED CACHE{PLUGIN_LIST} AND DEFINED CACHE{VISUALIZER_LIST_LIST})
+    if(DEFINED CACHE{LAST_YAML_BUILD} AND DEFINED CACHE{PLUGIN_LIST} AND DEFINED CACHE{VISUALIZER_LIST})
```

Comment:
Is it necessary to check the cache for all 3 variables? Or it is sufficient to just check for the ``LAST_YAML_BUILD``? If it's the latter case, I think it would be more digestible to have something like:
```
if(NOT DEFINED CACHE{LAST_YAML_BUILD})
    set(LAST_YAML_BUILD "0" CACHE INTERNAL "")
endif()
```

Reply:
I do not know about controlling the rebuilding of plugins. It comes down to how CMake decides something is stale. If the number of targets (e.g. plugins) changes, or their order changes it may just decide that everything is stale. I am not sure there is much we can do.

Regarding the check, yes we need to check all three because each is needed. There are ways that these entries could be removed/invalidated from the cache, so we need to be sure they are there.

Reply:
Got it. In that case, I think we should keep the ``message`` and have it say something like, "Using cached profiles" or whatever that should represent.

Reply:
Just pushed a fix which adds a message